src/model/channel/channel_id.rs

```diff
@@ -285,10 +285,18 @@ impl ChannelId {
 
     /// Search the cache for the channel with the Id.
     #[cfg(feature = "cache")]
+    #[deprecated(since = "0.5.8", note = "Use the `to_channel_cached`-method instead.")]
     pub fn find(&self) -> Option<Channel> { CACHE.read().channel(*self) }
 
+    /// Attempts to find a [`Channel`] by its Id in the cache.
+    ///
+    /// [`Channel`]: ../channel/enum.Channel.html
+    #[cfg(feature = "cache")]
+    pub fn to_channel_cached(self) -> Option<Channel> { CACHE.read().channel(self) }
+
     /// Search the cache for the channel. If it can't be found, the channel is
     /// requested over REST.
+    #[deprecated(since = "0.5.8", note = "Use the `to_channel`-method instead.")]
     pub fn get(&self) -> Result<Channel> {
         #[cfg(feature = "cache")]
```

Comment:
We can remove the implementations of the implemented methods and have them call the new methods to remove code duplication.